newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -0,0 +1,291 @@
+  const { loadedReceivedAssetPackInStore } = React.useContext(
+    AssetStoreContext
+  );
+  const [isBuying, setIsBuying] = React.useState(false);
+  const [
+    isCheckingPurchasesAfterLogin,
+    setIsCheckingPurchasesAfterLogin,
+  ] = React.useState(!receivedAssetPacks);
+  const [purchaseSuccessful, setPurchaseSuccessful] = React.useState(false);
+  const onContinue = async () => {
+    if (!profile) return;
+    try {
+      setIsBuying(true);
+      const checkoutUrl = await getStripeCheckoutUrl(getAuthorizationHeader, {
+        stripePriceId: privateAssetPackListingData.prices[0].stripePriceId,
+        userId: profile.id,
+        customerEmail: profile.email,
+      });
+      Window.openExternalURL(checkoutUrl);
+    } catch (error) {
+      console.error('Unable to get the checkout URL', error);
+      setIsBuying(false);
+      showErrorBox({
+        message: `Unable to get the checkout URL. Please try again later.`,
+        rawError: error,
+        errorId: 'asset-pack-checkout-error',
+      });
+    }
+  };
+
+  const checkUserPurchases = React.useCallback(
+    async () => {
+      if (!profile) return;
+      try {
+        const userPurchases = await listUserPurchases(getAuthorizationHeader, {
+          userId: profile.id,
+          productType: 'asset-pack',
+          role: 'receiver',
+        });
+        if (
+          userPurchases.find(
+            userPurchase =>
+              userPurchase.productId === privateAssetPackListingData.id
+          )
+        ) {
+          // We found the purchase, the user has bought the asset pack.
+          // We do not close the dialog yet, as we need to trigger a refresh of the asset store.
+          await onSuccessfulPurchase();
+        }
+      } catch (error) {
+        console.error('Unable to get the user purchases', error);
+        showErrorBox({
+          message: `Unable to retrieve your purchases. Please try again later.`,
+          rawError: error,
+          errorId: 'asset-pack-purchase-error',
+        });
+      }
+    },
+    [
+      profile,
+      getAuthorizationHeader,
+      privateAssetPackListingData,
+      onSuccessfulPurchase,
+    ]
+  );
+
+  useInterval(
+    () => {
+      checkUserPurchases();
+    },
+    isBuying ? 3900 : null
+  );
+
+  // Listen to the received asset pack, to know when a user has just logged in.
+  // In this case, start a timeout to remove the loader and give some time for the asset store to refresh.
+  React.useEffect(
+    () => {
+      let timeoutId;
+      (async () => {
+        if (receivedAssetPacks) {
+          timeoutId = setTimeout(
+            () => setIsCheckingPurchasesAfterLogin(false),
+            3000
+          );
+        }
+      })();
+      return () => {
+        clearTimeout(timeoutId);
+      };
+    },
+    [receivedAssetPacks]
+  );
+
+  // If the user has received this particular pack, either:
+  // - they just logged in, and already have it, so we close the dialog.
```

Comment:
Should we close a Dialog without user consent, It might give a feeling of loss of control. Especially when the user is considering spending money

Reply:
I agree with you.
But in this case, I think it's fine because it will not happen a lot, it's an edge case:
- the user already have the asset pack in their account
- they are not logged in
- they go to use it and see the buying page and decide to continue
- they are asked to log in
- we detect they already have it
- we close the dialog and they see the asset pack opened in the store

Reply:
Ok, fine for me!